### ISSUE TITLE
The "off" state of the LEDs are red using OutputLinear via wled udp

### ISSUE THREAD

**Issue by LeFloatyPancake** [cnlohr/colorchord#124]
Experimented with this a bit using ubuntu mate in a vm and wled version 0.12.0. The other outputs such cell, voronoi, and prominent work fine but the linear output has slight problems. 

In your YT vids showcasing what seems to be OutputLinear, all the LEDs turn off when no audio is being played. In my setup with 12 LEDs, they dip to a dim red when no audio is played. Is there some setting to turn them off? This maybe an issue in my end with pulseaudio loopback but irdk what the cause is.

I used the wled.conf config and did not really tweak it.

**Comment by CaiB** [comment]
When silent, linear should turn off the LEDs.

I’m not familiar with how PulseAudio works, but on WASAPI, there is (on some hardware) a difference between silence and no audio. When no audio is played, nothing is called and state does not get updated. Could you try to play some audio/video that has audio, but is completely silent? (Open Audacity and generate silence, or just find some silent YouTube video, etc) This should cause it to update the buffers in that case and help narrow down if it’s on the input or output side.

**Comment by LeFloatyPancake** [comment]
Tried playing music with some silence in between as well as the audacity trick. The LEDs are still stuck in red and do not turn off. I also tried setting the colors on WLED to black before starting colorchord but still the same issue.

**Comment by CaiB** [comment]
Try commenting out the two `wled_` lines near the bottom of the config to switch back to a plain network output, then adding these 2 lines:
```
skipfirst=2
firstval=2
```

Then put WLED into DRGB mode. You may also have to change the port ColorChord is sending to, not 100% sure where it expects DRGB data. Try 21324.

**Comment by CaiB** [comment]
Try commenting out the two `wled_` lines near the bottom of the config to switch back to a plain network output, then adding these 2 lines:
```
skipfirst=2
firstval=2
```

Then put WLED into DRGB mode. You may also have to change the port ColorChord is sending to, not 100% sure where it expects DRGB data. Try 21324.

**Comment by LeFloatyPancake** [comment]
Changed DMX Mode setting to single DRGB (this is the only one that says DRGB so idk) plus a restart. Also tried the default 21324 UDP port and the second port which is 65506. Same thing. The LEDs respond to music but dips to red during silent scenes.

Added the skipfirst and firstval to the config but it's still red. This got me wondering though if its an issue with the wled firmware since OutputCells can turn off leds during silent scenes.

**Comment by CaiB** [comment]
Yeah, sorry about that. I’ve never used WLED, just going off what someone else mentioned.  
If you’re willing to grab one, a packet captured from Wireshark would help check that there’s no red coming out of ColorChord.

The fact that it’s always red makes an audio input issue seem unlikely. I’m confused why there would be a difference between cells and linear though, since I thought those use the same sending logic… I’m not quite ready to blame WLED yet.

**Comment by LeFloatyPancake** [comment]
I tried messing around with skipfirst values with numbers 0-9. Quite interesting that it cycles through solid colors red, green, and blue but never black. Sticking with OutputCells for now until I find out whats missing in the config.

**Comment by CaiB** [comment]
Using DRGB is a bit of a hack just to test here. ColorChord will start a packet with `skipfirst` number of bytes filled with `firstval` as data, then just all the LEDs one after another after the “header”.

In DRGB mode WLED expects 2 bytes at the beginning of the packet, if I remember correctly, just a byte with the number 2 for DRGB and then an idle timeout byte, followed by the LED data.

Conveniently we can mush those 2 together to get an output for testing. By changing the `skipfirst` value you’re shifting around the location of the RGB bytes for the LEDs. The fact it changes colours tells me that the packet must contain a value on one of the colour channels, and by shifting it around you are seeing different colours. Sounds like something is wonky in ColorChord’s sending of linear data here…

Feel free to give the .NET version a try, I use a totally different sending architecture there. There’s an open issue about WLED with some sample configs to start from, but definitely let me know if you have any trouble/questions!

**Comment by CaiB** [comment]
Were you able to find anything more on this? I am curious what’s going on.

**Comment by LeFloatyPancake** [comment]
Haven't tried the .NET version yet...I had to keep this VM/ ARGB experimentation on hold as I've been quite busy lately. Hoping to get back to this project at a later time. Thanks for the help.
